lib/db/memory.js

```diff
@@ -260,27 +250,39 @@ MemoryStore.prototype = {
     }
 
     var activeClientIds = [];
-    var activeClients = [];
+    var activeClients = {};
     var ids = Object.keys(this.tokens);
     for (var i = 0; i < ids.length; i++) {
       var id = ids[i];
       if (this.tokens[id].userId.toString('hex') === uid) {
-        activeClientIds.push(this.tokens[id].clientId);
+        activeClientIds.push({
+          clientId: this.tokens[id].clientId,
+          clientTokenTime: this.tokens[id].createdAt
+        });
       }
     }
 
     // unique clients
-    activeClientIds.forEach(function (clientIdBuf) {
-      var clientIdHex = unbuf(clientIdBuf);
-      if (! findByBufProp(activeClients, 'id', clientIdHex)) {
-        var c = self.clients[clientIdHex];
-        if (c.canGrant === false) {
-          activeClients.push(c);
+    activeClientIds.forEach(function (clientBuf) {
+      var clientIdHex = unbuf(clientBuf.clientId);
+      var client = self.clients[clientIdHex];
+      if (client.canGrant === false) {
+        if (! activeClients[clientIdHex]) {
+          activeClients[clientIdHex] = client;
+        }
+        var clientTokenTime = unbuf(clientBuf.clientTokenTime);
+        if (clientTokenTime > activeClients[clientIdHex].createdAt) {
+          activeClients[clientIdHex].createdAt = clientTokenTime;
         }
       }
     });
 
-    var customSort = activeClients.slice(0);
+    var activeClientsArray =
```

Comment:
```
+    var activeClientsArray = Object.keys(activeClients).map(function (key) {
```

This can just be a single line, no need to line break after `=`